typescript/lib/parsers/palm.ts

```diff
@@ -0,0 +1,50 @@
+import { DiscussServiceClient } from "@google-ai/generativelanguage";
+import { JSONObject } from "../../common";
+import { Prompt, Output } from "../../types";
+import { AIConfigRuntime } from "../config";
+import { InferenceOptions } from "../modelParser";
+import { ParameterizedModelParser } from "../parameterizedModelParser";
+import { GoogleAuth } from "google-auth-library";
+import { getAPIKeyFromEnv } from "../utils";
+
+/**
+ * Model Parser for PaLM Text Generation. PaLM API Currently does not support streaming
+ * models/chat-bison-001
+ */
+
+export class PaLMTextParser extends ParameterizedModelParser {
+  _id = "models/text-bison-001";
```

Comment:
Nit: in future, add TODOs like `"Do not hardcode model"` so it's clear this will be addressed (or not addressed, but at least intent to address) later

Reply:
not a TODO, PaLM api only supports two models, we'll have one parser for each